src/packaging/metadata.py

```diff
@@ -633,6 +641,33 @@ def _process_requires_dist(
         else:
             return reqs
 
+    def _process_license_expression(self, value: str) -> str | None:
+        try:
+            return licenses.normalize_license_expression(value)
+        except ValueError as exc:
+            raise self._invalid_metadata(
+                f"{value!r} is invalid for {{field}}", cause=exc
+            ) from exc
+
+    def _process_license_files(self, value: list[str]) -> list[str]:
+        paths = []
+        for path in value:
+            if ".." in path:
+                raise self._invalid_metadata(
+                    f"{path!r} is invalid for {{field}}, "
+                    "parent directory indicators are not allowed"
+                )
+            if "*" in path:
+                raise self._invalid_metadata(
+                    f"{path!r} is invalid for {{field}}, " "paths must be resolved"
+                )
```

Comment:
I am not sure this is correct.

A literal `*` may be present in a resolved path and the PEP does not prohibit that.

Simple demo:

```bash
$ touch 'LICENSE***FILE'
$ LICENSE*
LICENSE***FILE
```

`LICENSE*` is a valid glob for pyproject.toml according to the PEP. Hence `LICENSE***FILE` (if it exists) is a valid value of `License-File` core metadata field.

It is likely not intended to have a literal `*` in a resolved path, more likely some backend made a mistake here. But I don't think packaging should do that kind of heuristic.

Reply:
FYI that is an invalid character on Windows.